src/QtDBMigration.cpp

```diff
@@ -43,10 +45,21 @@ class QtDBMigrationPrivate
     AbstractConfig *m_config;
 };
 
+void QtDBMigrationPrivate::setConfig(QString path) {
+    QFileInfo inputPath(path);
+    if (inputPath.isDir() && inputPath.exists()) {
+        m_config = new FolderConfig(path);
+    }
+    else  {
+        m_config = new JsonConfig(path);
+    }
+}
+
 QtDBMigrationPrivate::QtDBMigrationPrivate(QtDBMigration *q) :
-    m_q(q) ,
-    m_config(new JsonConfig(q->m_configPath))
+    m_q(q)
 {
+
+    setConfig(q->m_configPath);
```

Comment:
I don't like this.

This should be explicit! If you use constructor with `QDir`, it is explicitly set that you want to have a folder config.

One better option would be to introduce static methods: 
```
fromJson(filepath)
fromDir(path)
```

This way, it is too obscure.